--- core/dictgen/src/rootcling_impl.cxx
+++ core/dictgen/src/rootcling_impl.cxx
@@ -3615,6 +3615,11 @@
             llvm::cl::cat(gRootclingOptions),
             llvm::cl::init("-"));
+static llvm::cl::opt<std::string>
+gOptTarget("target", llvm::cl::Hidden,
+            llvm::cl::desc("Specify a target triple."),
+            llvm::cl::cat(gRootclingOptions),
+            llvm::cl::init("-"));
 static llvm::cl::list<std::string>
 gOptIncludePaths("I", llvm::cl::Prefix, llvm::cl::ZeroOrMore,
@@ -4044,6 +4049,14 @@
       clingArgs.push_back(gOptISysRoot.ValueStr.str());
    }
 
+   if (gOptTarget != "-") {
+      if (gOptTarget.empty()) {
+         ROOT::TMetaUtils::Error("", "-target specified without a value.\n");
+         return 1;
+      }
+      clingArgs.push_back("-target");
+      clingArgs.push_back(gOptTarget.getValue());
+   }
+
    for (const std::string &PPDefine : gOptPPDefines) {
       clingArgs.push_back(std::string("-D") + PPDefine);
    }
